### PR TITLE
Combine var readers + context extractor bug fix

### PR DESCRIPTION
**Pull request by maxaalexeeva**
- moved planting area rules to regular variable rules
- removed separate planting reader 
- removed path to grammar in the shell bc now there's only one 
- moved planting area tests to variable tests (label-based tests)
- fixed a couple of context extractor bugs + added a few context extractor tests

### CODE REVIEW

**Comment by maxaalexeeva** [comment]
@IkeKobby fyi

File: src/main/resources/variables/entities.yml
Comment (kwalcock) [inline src/main/resources/variables/entities.yml:48]: I don't know anything about this, but it seems strange that "sown" would be a lemma.  Wouldn't it be "sow"?

File: src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala
Comment (kwalcock) [inline src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala:58]: I like this use of a Map for this small number of tests.  If there were to be lots of them eventually, then the keys might be turned into constants/variables so that the compiler will check for typos or in case the name ever changes.  val DATE = "date".  If they all have the same keys, it could be a class instead, possibly with automatic defaults of missing values to "N/A".

File: src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala
Comment (kwalcock) [inline src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala:100]: It doesn't matter, but if one wanted to get rid of this return, one might use recursion on the windowSize.

**Review by kwalcock — APPROVED** [review]
LGTM if that sown is OK.

File: src/main/resources/variables/entities.yml
Comment (maxaalexeeva) [reply src/main/resources/variables/entities.yml:48]: Yes, absolutely should be sow. Thank you!

File: src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala
Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala:100]: I don't think I know how to do that :'(

File: src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala
Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala:66]: I did not completely understand about the class, but is what i did in this commit what you meant about variables? https://github.com/clulab/habitus/pull/137/commits/5277b4df279050e1dd7b3d119971183c8bcf44b5

Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala:66]: Yes.  If it was me, I would have made a typo in one of the keys already, but now the compiler will cover me.  The other way is something like

```scala
  case class Record(location: String = NA, date: String = NA, process: String = NA, crop: String = NA, fertilizer: String = NA, comparative: String = NA)

  Record(location = "Senegal", date = "2021/2022", process = "planting", comparative = "0")
```
which guarantees that the set of keys is fixed so that I can't add "oops" -> "a daisy" and the default value is automatic, .  However, it was mostly a thought experiment and I wouldn't change what's there without a better reason.

File: src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala
Comment (kwalcock) [reply src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala:100]: It's not important.  Some people have taken an oath never to use "return" and have thought about ways to avoid it.  If it changes some day, I'll tag you.

File: src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala
Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala:66]: I see! That does look nice...

Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala]: I know you said not to change it... But I changed it... It is just so much more beautiful this way!

Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala]: I have to learn to keep my mouth closed ;-)

Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala]: Nooo! I need to learn!! Thanks!!

Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala]: Does it look ok now? Should I merge?

Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala]: I was just waiting on Mihai since he was named as another reviewer and I wouldn't want to cross him.  Something in a different email hinted at a disagreement.  It's fine with me.

**Comment by maxaalexeeva** [comment]
@MihaiSurdeanu would you have a look at this pr? look ok to merge? Thanks!

**Comment by MihaiSurdeanu** [comment]
LGTM! Thanks for working on it and for being patient!

**Comment by kwalcock** [comment]
Merged!

**Comment by maxaalexeeva** [comment]
@MihaiSurdeanu thanks!

**Comment by maxaalexeeva** [comment]
@kwalcock thank you!

**Pull request merged**
